File: test/dotnet/TestOracle.cs
using Oracle.ManagedDataAccess.Client;

namespace DbTests;

// Oracle Database 接続テスト（ODP.NET Core / Oracle.ManagedDataAccess.Core）。
// マネージド版のため Oracle Client（Instant Client）のインストールは不要。
public static class TestOracle
{
    public static async Task RunAsync()
    {
        Console.WriteLine("===== Test Oracle =====");
        await using var conn = new OracleConnection(Config.Oracle);
        await conn.OpenAsync();

        await using var cmd = new OracleCommand("SELECT * FROM Shippers", conn);
        await using var reader = await cmd.ExecuteReaderAsync();
        await Table.PrintReaderAsync(reader);
    }
}

File: test/python/test_oracle.py
"""Oracle Database 接続テスト（python-oracledb / Thin モード）。

Thin モードが既定のため、Oracle Client（Instant Client）のインストールは不要。
"""
import oracledb

from config import CONNECT_TIMEOUT_SEC, ORACLE
from util import print_table


def run():
    print("===== Test Oracle =====")
    conn = oracledb.connect(
        user=ORACLE["user"],
        password=ORACLE["password"],
        dsn=ORACLE["dsn"],
        tcp_connect_timeout=CONNECT_TIMEOUT_SEC,
    )
    try:
        with conn.cursor() as cur:
            cur.execute("SELECT * FROM Shippers")
            rows = cur.fetchall()
            columns = [d[0] for d in cur.description]
            print_table(columns, rows)
    finally:
        conn.close()


if __name__ == "__main__":
    run()
